### PR TITLE
Elasticsearch: Add error source for DataQuery

### PR DESCRIPTION
**Pull request by ivanahuckova**
In this PR, we are adding logic to support `status_source` for `DataQuery`. We are using methods and http client from freshly released `github.com/grafana/grafana-plugin-sdk-go v0.188.4`, so we need to also update that package.

### CODE REVIEW

File: pkg/tsdb/elasticsearch/data_query.go [hunk 1]
```diff
@@ -41,12 +42,15 @@ var newElasticsearchDataQuery = func(ctx context.Context, client es.Client, data
...
+	response := backend.NewQueryDataResponse()
 	e.logger.Debug("Parsing queries", "queriesLength", len(e.dataQueries))
 	queries, err := parseQuery(e.dataQueries, e.logger)
 	if err != nil {
 		mq, _ := json.Marshal(e.dataQueries)
 		e.logger.Error("Failed to parse queries", "error", err, "queries", string(mq), "queriesLength", len(queries), "duration", time.Since(start), "stage", es.StagePrepareRequest)
-		return &backend.QueryDataResponse{}, err
+		err := errorsource.PluginError(err, false)
+		response.Responses[e.dataQueries[0].RefID] = errorsource.Response(err)
+		return response, err
```
Comment (scottlepp) [inline pkg/tsdb/elasticsearch/data_query.go:53]: looks like we are repeating ourselves here.  maybe just create an `assignSource` function

```
func assignSource(refID string, response backend.QueryDataResponse, err error) (backend.QueryDataResponse, error) {
  err := errorsource.PluginError(err, false)
  response.Responses[refID] = errorsource.Response(err)
  return response, err
```

then just 
```
return assignSource(e.dataQueries[0].RefID, response, err)
```

File: pkg/tsdb/elasticsearch/data_query.go [hunk 2]
```diff
@@ -493,3 +497,9 @@ func stringToIntWithDefaultValue(valueStr string, defaultValue int) int {
 	}
 	return value
 }
+
+func addPluginErrorToResponse(refID string, response *backend.QueryDataResponse, err error) *backend.QueryDataResponse {
```
Comment (scottlepp) [inline pkg/tsdb/elasticsearch/data_query.go:501]: [pr](https://github.com/grafana/grafana-plugin-sdk-go/pull/795/files#diff-cf82f3baeb1eb237f38b3190af31d740586f6fbf3db8180317719dee4fa4c184R79) to add this function to sdk

File: pkg/tsdb/elasticsearch/elasticsearch.go [hunk 3]
```diff
@@ -78,6 +79,7 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
 			return nil, fmt.Errorf("error reading settings: %w", err)
 		}
 		httpCliOpts, err := settings.HTTPClientOptions(ctx)
+		httpCliOpts.Middlewares = append(httpCliOpts.Middlewares, errorsource.Middleware("elasticsearch"))
```
Comment (scottlepp) [inline pkg/tsdb/elasticsearch/elasticsearch.go:82]: added a [New constructor](https://github.com/grafana/grafana-plugin-sdk-go/pull/795/files#diff-bebb0be07bf4d976fc20540020d140a95476dcd59431154dcc6aebcc1048343dR12) for http client.  so you don't have to explicitly add the middleware

File: pkg/tsdb/elasticsearch/data_query.go [hunk 4]
```diff
@@ -57,21 +59,23 @@ func (e *elasticsearchDataQuery) execute() (*backend.QueryDataResponse, error) {
...
+		return errorsource.AddPluginErrorToResponse(e.dataQueries[0].RefID, response, err), err
 	}
 
 	e.logger.Info("Prepared request", "queriesLength", len(queries), "duration", time.Since(start), "stage", es.StagePrepareRequest)
 	res, err := e.client.ExecuteMultisearch(req)
 	if err != nil {
-		return &backend.QueryDataResponse{}, err
+		// We are returning error containing the source that was added trough errorsource.Middleware
+		response.Responses[e.dataQueries[0].RefID] = errorsource.Response(err)
+		return response, err
```
Comment (scottlepp) [inline pkg/tsdb/elasticsearch/data_query.go:78]:
```suggestion
        return errorsource.AddErrorToResponse(e.dataQueries[0].RefID, response, err), err
```

**Review by scottlepp — APPROVED** [review]
LGTM - added a suggestion

File: pkg/tsdb/elasticsearch/data_query.go [hunk 5]
```diff
@@ -41,12 +42,13 @@ var newElasticsearchDataQuery = func(ctx context.Context, client es.Client, data
...
 func (e *elasticsearchDataQuery) execute() (*backend.QueryDataResponse, error) {
 	start := time.Now()
+	response := backend.NewQueryDataResponse()
 	e.logger.Debug("Parsing queries", "queriesLength", len(e.dataQueries))
 	queries, err := parseQuery(e.dataQueries, e.logger)
 	if err != nil {
 		mq, _ := json.Marshal(e.dataQueries)
 		e.logger.Error("Failed to parse queries", "error", err, "queries", string(mq), "queriesLength", len(queries), "duration", time.Since(start), "stage", es.StagePrepareRequest)
-		return &backend.QueryDataResponse{}, err
+		return errorsource.AddPluginErrorToResponse(e.dataQueries[0].RefID, response, err), nil
```
Comment (svennergr) [inline pkg/tsdb/elasticsearch/data_query.go:51]: Is it correct to not return an error in the second return value anymore? If so, we could get rid of the definition in line 43 as well, and remove tests in e.g. `data_query_test.go` line 31.

Comment (ivanahuckova) [reply pkg/tsdb/elasticsearch/data_query.go:51]: Will be resolved in ext PR when we refactor `parseResponse`

**Pull request merged**